spec/integration/type/tidy_spec.rb

```diff
@@ -11,6 +11,28 @@
     Puppet::Util::Storage.stubs(:store)
   end
 
+  it "should be able to recursively remove directories" do
+    dir = tmpfile("tidy_testing")
+    foo = File.join(dir, "foo")
+    bar = File.join(foo, "bar")
+
+    Dir.mkdir(dir)
+    Dir.mkdir(foo)
+    Dir.mkdir(bar)
+
+    tidy = Puppet::Type.type(:tidy).new :path => dir, :recurse => true, :rmdirs => true
+
+    catalog = Puppet::Resource::Catalog.new
+    catalog.add_resource(tidy)
+    # avoid crude failures because of nil resources that result
+    # from implicit containment and lacking containers
+    catalog.stubs(:container_of).returns tidy
```

Comment:
You might be able to use `apply_compiled_manifest` like we do in https://github.com/puppetlabs/puppet/blob/master/spec/unit/transaction_spec.rb#L640-L645 instead. That way you don't have to stub implementation details.